coretex/entities/sample/network_sample.py

```diff
@@ -112,6 +114,31 @@ def _createSample(
 
         return cls.decode(response.json)
 
+    def modifiedSinceLastDownload(self) -> bool:
+        """
+            Checking if sample has been modified since last download, if the sample is already
+            stored locally
+
+            Returns
+            -------
+            bool -> False if sample has not changed since last download, True otherwise
+        """
+
+        if not self.zipPath.exists():
+            return False
+
+        sampleJson = self.fetchById(self.id).encode()
+        lastModified = sampleJson["storage_last_modified"]
```

Comment:
Why is the sample being fetched and encoded here?

Reply:
The needed field "storage_last_modified" gets sent only for fetch requests (as far as I could determine). It gets encoded afterward to read the "storage_last_modified" field from the original request json

Reply:
Sample should not be fetched again. Update implementation to not rely on fetching new samples